mono/metadata/locales.c

```diff
@@ -294,20 +300,20 @@ construct_culture (MonoCultureInfo *this_obj, const CultureInfoEntry *ci)
 }
 
 static MonoBoolean
-construct_region (MonoRegionInfo *this_obj, const RegionInfoEntry *ri)
+construct_region (MonoRegionInfoHandle* this_obj, const RegionInfoEntry *ri)
 {
 	MonoDomain *domain = mono_domain_get ();
 
-	this_obj->geo_id = ri->geo_id;
-	MONO_OBJECT_SETREF (this_obj, iso2name, mono_string_new (domain, idx2string (ri->iso2name)));
-	MONO_OBJECT_SETREF (this_obj, iso3name, mono_string_new (domain, idx2string (ri->iso3name)));
-	MONO_OBJECT_SETREF (this_obj, win3name, mono_string_new (domain, idx2string (ri->win3name)));
-	MONO_OBJECT_SETREF (this_obj, english_name, mono_string_new (domain, idx2string (ri->english_name)));
-	MONO_OBJECT_SETREF (this_obj, native_name, mono_string_new (domain, idx2string (ri->native_name)));
-	MONO_OBJECT_SETREF (this_obj, currency_symbol, mono_string_new (domain, idx2string (ri->currency_symbol)));
-	MONO_OBJECT_SETREF (this_obj, iso_currency_symbol, mono_string_new (domain, idx2string (ri->iso_currency_symbol)));
-	MONO_OBJECT_SETREF (this_obj, currency_english_name, mono_string_new (domain, idx2string (ri->currency_english_name)));
-	MONO_OBJECT_SETREF (this_obj, currency_native_name, mono_string_new (domain, idx2string (ri->currency_native_name)));
+	HANDLE_SET_VAL (this_obj, geo_id, ri->geo_id);
+	HANDLE_SET_MP (this_obj, iso2name, mono_string_new (domain, idx2string (ri->iso2name)));
```

Comment:
I added a MONO_HANDLE_SETREF (https://github.com/mono/mono/pull/2314/files#diff-45ebb3d3ae14abc6bf0815c343fe0cd4R86)  and MONO_HANDLE_SET (https://github.com/mono/mono/pull/2314/files#diff-45ebb3d3ae14abc6bf0815c343fe0cd4R94)